src/Poseidon/Janno.hs

```diff
@@ -1037,23 +1082,27 @@ checkJannoConsistency jannoPath xs
 checkIndividualUnique :: JannoRows -> Bool
 checkIndividualUnique (JannoRows rows) = length rows == length (nub $ map jPoseidonID rows)
 
-checkJannoRowConsistency :: FilePath -> Int -> JannoRow -> Either PoseidonException JannoRow
-checkJannoRowConsistency jannoPath row x
-    | not $ checkMandatoryStringNotEmpty x = Left $ PoseidonFileRowException jannoPath row
-          "The mandatory columns Poseidon_ID and Group_Name contain empty values"
-    | not $ checkC14ColsConsistent x = Left $ PoseidonFileRowException jannoPath row
-          "The Date_* columns are not consistent"
-    | not $ checkContamColsConsistent x = Left $ PoseidonFileRowException jannoPath row
-          "The Contamination_* columns are not consistent"
-    | not $ checkRelationColsConsistent x = Left $ PoseidonFileRowException jannoPath row
-          "The Relation_* columns are not consistent"
-    | otherwise = Right x
-
-checkMandatoryStringNotEmpty :: JannoRow -> Bool
+-- Row-wise janno consistency checks
+
+type JannoRowWarnings = [String]
+type JannoRowLog = E.ExceptT String (W.Writer JannoRowWarnings)
```

Comment:
As discussed, I think I would generally prefer the policy "If you want to log, you need PoseidonIO". Inventing a new Monad-Stack that preserves purity in such a narrow area is a bit of an overkill in my view. I think a Writer would make sense if we were to collect log-messages over much longer, transcending rows and even files, but for a single row? Anyway, as discussed, I have understood the code and I'm OK with it. Also, this kind of code might be useful in the future, so perhaps good to have a template that already works.

Reply:
Your point is well taken :+1: